src/sage/matroids/chow_ring_ideal.py

```diff
@@ -0,0 +1,643 @@
+        MPolynomialIdeal.__init__(self, poly_ring, self._gens_constructor(poly_ring))
+
+    def _gens_constructor(self, poly_ring):
+        r"""
+        Return the generators of ``self``.
+
+        EXAMPLES::
+
+            sage: ch = matroids.Wheel(3).chow_ring(QQ, True, 'fy')
+            sage: ch.defining_ideal()._gens_constructor(ch.defining_ideal().ring())
+            [B0*B1, B0*B2, B0*B3, B0*B4, B0*B5, B0*B124, B0*B15, B0*B23,
+             B0*B345, B0*B1, B1*B2, B1*B3, B1*B4, B1*B5, B1*B025, B1*B04,
+             B1*B23, B1*B345, B0*B2, B1*B2, B2*B3, B2*B4, B2*B5, B2*B013,
+             B2*B04, B2*B15, B2*B345, B0*B3, B1*B3, B2*B3, B3*B4, B3*B5,
+             B3*B025, B3*B04, B3*B124, B3*B15, B0*B4, B1*B4, B2*B4, B3*B4,
+             B4*B5, B4*B013, B4*B025, B4*B15, B4*B23, B0*B5, B1*B5, B2*B5,
+             B3*B5, B4*B5, B5*B013, B5*B04, B5*B124, B5*B23, B2*B013, B4*B013,
+             B5*B013, B013*B025, B013*B04, B013*B124, B013*B15, B013*B23,
+             B013*B345, B1*B025, B3*B025, B4*B025, B013*B025, B025*B04,
+             B025*B124, B025*B15, B025*B23, B025*B345, B1*B04, B2*B04, B3*B04,
+             B5*B04, B013*B04, B025*B04, B04*B124, B04*B15, B04*B23, B04*B345,
+             B0*B124, B3*B124, B5*B124, B013*B124, B025*B124, B04*B124,
+             B124*B15, B124*B23, B124*B345, B0*B15, B2*B15, B3*B15, B4*B15,
+             B013*B15, B025*B15, B04*B15, B124*B15, B15*B23, B15*B345, B0*B23,
+             B1*B23, B4*B23, B5*B23, B013*B23, B025*B23, B04*B23, B124*B23,
+             B15*B23, B23*B345, B0*B345, B1*B345, B2*B345, B013*B345,
+             B025*B345, B04*B345, B124*B345, B15*B345, B23*B345, A0*B, A0*B1,
+             A0*B2, A0*B3, A0*B4, A0*B5, A0*B124, A0*B15, A0*B23, A0*B345,
+             A1*B, A1*B0, A1*B2, A1*B3, A1*B4, A1*B5, A1*B025, A1*B04, A1*B23,
+             A1*B345, A2*B, A2*B0, A2*B1, A2*B3, A2*B4, A2*B5, A2*B013, A2*B04,
+             A2*B15, A2*B345, A3*B, A3*B0, A3*B1, A3*B2, A3*B4, A3*B5, A3*B025,
+             A3*B04, A3*B124, A3*B15, A4*B, A4*B0, A4*B1, A4*B2, A4*B3, A4*B5,
+             A4*B013, A4*B025, A4*B15, A4*B23, A5*B, A5*B0, A5*B1, A5*B2,
+             A5*B3, A5*B4, A5*B013, A5*B04, A5*B124, A5*B23,
+             B + B0 + B1 + B2 + B3 + B4 + B5 + B013 + B025 + B04 + B124 + B15 + B23 + B345 + B012345,
+             A0 + B0 + B013 + B025 + B04 + B012345,
+             A1 + B1 + B013 + B124 + B15 + B012345,
+             A2 + B2 + B025 + B124 + B23 + B012345,
+             A3 + B3 + B013 + B23 + B345 + B012345,
+             A4 + B4 + B04 + B124 + B345 + B012345,
+             A5 + B5 + B025 + B15 + B345 + B012345]
+        """
+        E = list(self._matroid.groundset())
+        flats_containing = {x: [] for x in E}
+        for F in self._flats:
+            for x in F:
+                flats_containing[x].append(F)
+
+        Q = list()
+        L = list()
+        term = poly_ring.zero()
+        for F in self._flats:
```

Comment:
There is an antichain (by size) iterator for the lattice of flats. That will probably be faster.